src/main/java/microsoft/exchange/webservices/data/ServiceRequestBase.java

```diff
@@ -368,13 +356,85 @@ private static InputStream getResponseErrorStream(HttpWebRequest request)
   /**
    * Reads the response.
    *
+   * @return serviceResponse
+   * @throws Exception
+   */
+  protected T readResponse(HttpWebRequest response) throws Exception {
+    T serviceResponse;
+
+    if (!response.getResponseContentType().startsWith("text/xml")) {
+      String line = new BufferedReader(new InputStreamReader(ServiceRequestBase.getResponseStream(response)))
+          .readLine();
+      throw new ServiceRequestException(Strings.ServiceResponseDoesNotContainXml);
+    }
+
+    /**
+     * If tracing is enabled, we read the entire response into a
+     * MemoryStream so that we can pass it along to the ITraceListener. Then
+     * we parse the response from the MemoryStream.
+     */
+
+    try {
+      this.getService().processHttpResponseHeaders(
+          TraceFlags.EwsResponseHttpHeaders, response);
+
+      if (this.getService().isTraceEnabledFor(TraceFlags.EwsResponse)) {
+        ByteArrayOutputStream memoryStream = new ByteArrayOutputStream();
+        InputStream serviceResponseStream = ServiceRequestBase
+            .getResponseStream(response);
+        while (true) {
+          int data = serviceResponseStream.read();
```

Comment:
interchanged the while(true) it was previously just copy paste from the original one.